hive/hive-export-import-metastore.md

```diff
@@ -23,73 +23,98 @@ ms.author: dkakadia
-
-1) [Connect to HDInsight Cluster using SSH](https://docs.microsoft.com/en-us/azure/hdinsight/hdinsight-hadoop-linux-use-ssh-unix)
+Need to export Hive metastore and import it on another HDInsight cluster.
+
+* See [apache-hive-migrate-workloads](https://docs.microsoft.com/en-us/azure/hdinsight/interactive-query/apache-hive-migrate-workloads) for migration of an external metastore, or for upgrading Hive from HDInsight 3.6 to 4.0.
+
+* This current article gives instructions on using a script to export/import contents of an internal Hive metastore.
+
+* Import works only if the destination cluster shares the same Storage Account as the source cluster.
+
+> [!NOTE]
+>
+> * The script supports copying of tables and partitions. For HDInsight 4.0, it additionally covers constraints, views, and materialized views. Other metadata objects, like UDFs, must be copied manually.
+>
+> * All managed tables will become transactional if the output HDInsight version is 4.0. Optionally, make the table non-transactional by exporting the data to an external table with the property 'external.table.purge'='true'. For example,
+>
+>    ```SQL
+>    create table tablename_backup like tablename;
+>    insert overwrite table tablename_backup select * from tablename;
+>    create external table tablename_tmp like tablename;
+>    insert overwrite table tablename_tmp select * from tablename;
+>    alter table tablename_tmp set tblproperties('external.table.purge'='true');
+>    drop table tablename;
+>    alter table tablename_tmp rename to tablename;
+>
+> * This procedure preserves non-ACID table locations. You can manually edit the DDL in
+`alltables.hql`, generated from the script, to reflect any location changes. *For ACID tables, a new copy of the data will be created*.
+>
+> * The procedure assumes that after completion, the old cluster will **not** be used any longer.
+
+#### Resolution Steps:
+
+If migrating from an external metastore, follow steps in [apache-hive-migrate-workloads](https://docs.microsoft.com/en-us/azure/hdinsight/interactive-query/apache-hive-migrate-workloads). Otherwise, follow steps below.
+
+1) [Connect to the HDInsight Cluster using SSH](https://docs.microsoft.com/en-us/azure/hdinsight/hdinsight-hadoop-linux-use-ssh-unix), to the primary headnode.
+
+1) Download the export script from an ssh session to the cluster:
+
+    ```bash
+    wget "https://hdiconfigactions.blob.core.windows.net/hivemetastoreschemaupgrade/$SCRIPT"
+    chmod 755 "$SCRIPT"
+    ```
+
+    where `SCRIPT="exporthive_hdi_3_6.sh"` for HDInsight 3.6 or `SCRIPT="exporthive_hdi_4_0.sh"` for HDInsight 4.0.
+
+1) Run the script from the cluster:
+
+    * For HDInsight 4.0, follow these additional steps, first:
+
+        a. Download a helper script used by `exporthive_hdi_4_0.sh`.
+
+        ```bash
+        wget "https://hdiconfigactions.blob.core.windows.net/hivemetastoreschemaupgrade/constraints2altertable.py"
+        chmod 755 constraints2altertable.py
```

Comment:
If we downloaded a helper script, it could do the chmod on the customer's behalf too